sdk/dapp-kit/src/actions/connectWallet.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type {
+	StandardConnectInput,
+	WalletAccount,
+	WalletWithSuiFeatures,
+} from '@mysten/wallet-standard';
+import type { Dispatch } from 'react';
+import type { WalletAction } from '../reducers/walletReducer.js';
+import type { StorageAdapter } from '../utils/storageAdapters.js';
+import { setMostRecentWalletConnectionInfo } from '../utils/walletUtils.js';
+
+export type ConnectWalletArgs = {
+	/** The wallet to connect to. */
+	wallet: WalletWithSuiFeatures;
+
+	/** An optional account address to connect to. Defaults to the first authorized account. */
+	accountAddress?: string;
+} & StandardConnectInput;
+
+export async function connectWallet(
+	dispatch: Dispatch<WalletAction>,
+	storageAdapter: StorageAdapter,
+	storageKey: string,
+	{ wallet, accountAddress, ...standardConnectInput }: ConnectWalletArgs,
+) {
+	dispatch({ type: 'wallet-connection-status-updated', payload: 'connecting' });
```

Comment:
Still unconvinced that we need the status.

Reply:
Yeah not sure if you saw my comment in https://github.com/MystenLabs/sui/pull/13661/files#r1320291269, but I think there are a few valid use cases where you would want this state outside the context of using the connect/disconnect mutation hooks. If you still feel strongly after this, I'm glad to remove it haha

One potential use case would be if you wanted to disable a button for signing a transaction if your wallet isn't connected. A developer could obviously change the button to say "connect wallet" and then use the connect mutation state, but then we're enforcing constraints that don't give folks to build the flexibility for what they want 🤔